<?xml version="1.0"?>

<launch>
  <arg name="map_file" default="$(find icart_mini_navigation)/maps/mymap.yaml"/>
  <arg name="filename" default="$(find icart_mini_navigation)/waypoints_cfg/waypoints.yaml"/>

  <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)"/>
  <node name="waypoints_saver" pkg="waypoints_navigation" type="waypoints_saver">
    <param name="filename" value="$(arg filename)"/> 
    <remap from="waypoints_viz" to="clicked_point"/>
    <remap from="finish_pose" to="move_base_simple/goal"/>
  </node>
  
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find icart_mini_navigation)/rviz_cfg/record_waypoints.rviz"/>
</launch>
